x/auth/middleware/fee.go

```diff
@@ -62,7 +72,10 @@ func (txh mempoolFeeTxHandler) CheckTx(ctx context.Context, tx sdk.Tx, req abci.
 		}
 	}
 
-	return txh.next.CheckTx(ctx, tx, req)
+	res, err := txh.next.CheckTx(ctx, tx, req)
+	res.Priority = GetTxPriority(feeCoins)
```

Comment:
I think this should be implemented as a middleware. We may have different priority options and app can chose which one to use.

Reply:
If you can come up with a reasonable different option apart from what I've implemented, then I think that could make sense. I don't see merit in defining it as middleware if we just have 1 option.

Reply:
Actually, I can think of other reasons why middleware could be useful in this context. I'll update it.

Reply:
Few options:
1. priority by tx sender
2. priority by tx type: evidence > ibc > everything else (by fee)
3. osmosis could prioritize LP transactions over swaps.